core/org/commcare/core/process/CommCareInstanceInitializer.java

```diff
@@ -87,15 +99,23 @@ public AbstractTreeElement generateRoot(ExternalDataInstance instance) {
                 throw new RuntimeException("Could not load fixture for src: " + ref);
             }
         }
-        if (instance.getReference().contains("session")) {
+        if (instance.getReference().indexOf("session") != -1) {
             if(this.mPlatform == null) {
                 throw new RuntimeException("Cannot generate session instance with undeclared platform!");
             }
             User u = mSandbox.getLoggedInUser();
-            TreeElement root = session.getSessionInstance("----", "CommCare CLI: " + mPlatform.getMajorVersion() + "." + mPlatform.getMinorVersion(), u.getUsername(), u.getUniqueId(), u.getProperties()).getRoot();
+            TreeElement root = session.getSessionInstance(getDeviceId(), getVersionString(), u.getUsername(), u.getUniqueId(), u.getProperties()).getRoot();
             root.setParent(instance.getBase());
             return root;
         }
         return null;
     }
+
+    protected String getDeviceId(){
+        return "----";
+    }
+
+    protected String getVersionString(){
+        return "CommCare CLI: " + mPlatform.getMajorVersion() + "." + mPlatform.getMinorVersion();
```

Comment:
This is outside of the CLI code now and into core, right? Maybe we should accept the version string and device ID as arguments and/or continue keeping a CLI specific version that overrides these as abstract methods?